From 5a438fbb5b0d5821844a7e79a5fe88339a6b43c5 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Fri, 18 Aug 2023 17:07:28 +0530
Subject: [PATCH]  Added sensor_min & sensor_max values to sdr record, sdr raw 
 command is getting max and min sensor reading values now

Tested:

root@AMIOT-b2be60439ec2:~# ipmitool raw 0x0a 0x23 0x01 0x00 0x29 0x00 0x00 0xff
 ff ff 29 00 51 01 3b 20 00 29 00 01 00 68 01 01
 85 32 85 32 1b 1b 80 01 00 00 01 00 00 00 00 00
 00 00 00 00 7f 80 00 73 6e 00 00 05 00 00 00 00
 00 c8 42 4d 43 20 54 65 6d 70 00 00 00 00 00 00
 00 00
root@AMIOT-22872d483974:~# ipmitool raw 0x0a 0x23 0x02 0x00 0x18 0x00 0x00 0xff
 19 00 18 00 51 01 3b 20 00 18 00 01 00 68 04 01
 05 30 05 00 03 03 00 12 00 00 62 00 00 00 00 00
 00 00 00 00 a8 00 00 00 00 00 0b 0d 00 00 00 00
 00 c5 46 61 6e 20 32 00 00 00 00 00 00 00 00 00
 00 00
root@AMIOT-b2be60439ec2:~# ipmitool raw 0x0a 0x23 0x0c 0x00 0x45 0x00 0x00 0xff
 46 00 45 00 51 01 3b 20 00 45 00 01 00 68 02 01
 00 00 00 00 00 00 00 04 00 00 76 00 00 00 00 e0
 00 00 00 00 2c 00 00 00 00 00 00 00 00 00 00 00
 00 d0 50 53 55 31 20 49 6e 20 56 6f 6c 74 61 67
 65 00

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 src/sensorcommands.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/sensorcommands.cpp b/src/sensorcommands.cpp
index 6275ef7..d6ca999 100644
--- a/src/sensorcommands.cpp
+++ b/src/sensorcommands.cpp
@@ -1742,6 +1742,10 @@ static int
         min = std::visit(VariantToDoubleVisitor(), minObject->second);
     }
 
+   record.body.sensor_max = max;
+   record.body.sensor_min = min;
+
+
     int16_t mValue = 0;
     int8_t rExp = 0;
     int16_t bValue = 0;
-- 
2.25.1

